.gitignore

```diff
@@ -13,8 +13,6 @@ dist-ssr
 *.local
 
 # Editor directories and files
-.vscode/*
```

Comment:
I think it makes sense to ignore .vscode folder. If you really need to share some settings, make an exception for the specific files.
@marilari88 wdyt? are those files needed?

Reply:
That's an opinionated choice I made, but saying "ignore the score folder but not some files" doesn't make much sense. Either you keep the folder with its files or exclude it.

Here I'm enforcing two things basically: one is to set the proper formatter for the project so that if we have external contributors they don't push unformatted code, the second is providing a prompt for the user using vscode to install the right extension https://tattoocoder.com/recommending-vscode-extensions-within-your-open-source-projects/

Reply:
Anything that can help potential contributors write code is welcome. We will add the nth stuff to the root of the repo, but I personally don't care about it. @giacomocerquone you can leave .gitignore as you changed it

Reply:
The issue with the removal of .vscode folder from .gitignore is that there might be additional files in a local project configuration which should not be committed (e.g. tasks.json or launch.json).

```
.vscode/*
!.vscode/settings.json
``` 

Even this approach is not safe: if someone changes an option in the project locally which edits settings.json, it would be included with `git add .` and might be accidentally committed, but still better than no protection at all.

I've tried also the force add approach (ref. https://stackoverflow.com/questions/8006393/force-add-despite-the-gitignore-file), the idea was to keep the folder ignored and force add the file.
Unfortunately, further changes to the file are still tracked, so the result is the same as above.

Saw also https://stackoverflow.com/questions/4348590/how-can-i-make-git-ignore-future-revisions-to-a-file but index changes are kept locally.

Unfortunately there's no way (afaik at least) to use different file names in vscode to share settings and avoid the problem above, such as a different file with a base configuration extended by settings.json, e.g.
```
# .vscode
settings_base.json # needs to be edited manually, can be shared
settings.json      # developer custom stuff, must be ignored
```

Reply:
Yeah I mean I understand the problem, but as long as maintainers disallow insertion of new things in the folder what's the problem? Even allowing only certain files as you said don't solve the issue and excluding everything is worse.

Up to you though, tell me what you want

Reply:
(btw there are also GitHub bots that disallow merging when certain files are modified)

Reply:
>but as long as maintainers disallow insertion of new things in the folder what's the problem?

Maintainers makes mistakes too :sweat_smile: 
Ok as it is for now.

Reply:
Again, there are bots that prevents the merging of the pr

Reply:
Implies that even minor changes from mainteiners must require a PR.
I'm against that.

Eventually if something will be accidentally committed we'll fix that, is ok as it is until we'll find a good solution to prevent accidents.

Reply:
No, maintainers can be excluded from the rule (you can allow only certain people tu push changes to only certain files) furthermore maintainers can also avoid creating ors and push directly on master.

Your choice

Reply:
But we can make mistakes and commit stuff accidentally too, so that will not solve the problem.
It's fine as it is, it just wanted to start the discussion to check if that was necessary and explore alternatives.